File: ms_consulta/src/main/java/com/saude_mais/ms_consulta/entities/Consulta.java
package com.saude_mais.ms_consulta.entities;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Consulta {

    private String id;
    private String nomePac;
    private String nomeMed;
    private String dtColsulta;
    private Action action = Action.NONE;
    
}

File: validationSchema/ms_gerenciamento/consulta.js
use clinica;

db.createCollection("consulta", {
    validator: {
        $jsonSchema: {
            required: ["nomePac", "nomeMed", "dtColsulta", "action"],
            properties: {
                nomePac: {
                    bsonType: "string",
                    description: "nome do paciente"
                },
                nomeMed: {
                    bsonType: "string",
                    maxLength: 8,
                    description: "nome medico"
                },
                dtColsulta: {
                    bsonType: "string",
                    maxLength: 10,
                    description: "data da consulta"
                },
                action: {
                    bsonType: ["object"],
                    description: "status da consulta"
                }
            }
        }
    }
})

